src/DDTrace/Integrations/Lumen/LumenSandboxedIntegration.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace DDTrace\Integrations\Lumen;
+
+use DDTrace\GlobalTracer;
+use DDTrace\SpanData;
+use DDTrace\Integrations\SandboxedIntegration;
+use DDTrace\Tag;
+use DDTrace\Type;
+
+/**
+ * Lumen Sandboxed integration
+ */
+class LumenSandboxedIntegration extends SandboxedIntegration
+{
+    const NAME = 'lumen';
+
+    /**
+     * @return string The integration name.
+     */
+    public function getName()
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function requiresExplicitTraceAnalyticsEnabling()
+    {
+        return false;
+    }
+
+    /**
+     * @return int
+     */
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return SandboxedIntegration::NOT_LOADED;
+        }
+
+        $rootScope = GlobalTracer::get()->getRootScope();
+        $rootSpan = null;
+
+        if (null === $rootScope || null === ($rootSpan = $rootScope->getSpan())) {
+            return SandboxedIntegration::NOT_LOADED;
+        }
+
+        $integration = $this;
+        $appName = \ddtrace_config_app_name(self::NAME);
+
+        \dd_trace_method(
+            'Laravel\Lumen\Application',
+            'prepareRequest',
+            function (SpanData $span, $args) use ($rootSpan, $integration, $appName) {
+                $request = $args[0];
+                $rootSpan->overwriteOperationName('lumen.request');
+                $rootSpan->setTag(Tag::SERVICE_NAME, $appName);
+                $rootSpan->setIntegration($integration);
+                $rootSpan->setTraceAnalyticsCandidate();
+                $rootSpan->setTag(Tag::HTTP_URL, $request->getUri());
+                $rootSpan->setTag(Tag::HTTP_METHOD, $request->getMethod());
+                return false;
+            }
+        );
+
+        // Extracting resource name as in legacy integration
+        \dd_trace_method(
+            'Laravel\Lumen\Application',
+            'handleFoundRoute',
+            [
+                'prehook' => function (SpanData $span, $args) use ($rootSpan) {
```

Comment:
Prehook isn't supported on PHP 5. It looks like [Lumen 5.2 supports PHP 5.5+](https://lumen.laravel.com/docs/5.2). But we're only running the Lumen web tests for PHP 7.1+. It looks like Lumen is [supported on PHP 5 until Lumen 5.4](https://lumen.laravel.com/docs/5.4). I'm not sure using a posthook here would work without creating orphan spans. Maybe we only load the sandbox version on PHP 7 until we're able to get PHP 5 up to feature parity?

Also - it might be good to add `lumen-52-*` to `test-web-70` in `composer.json` since it is supported. :)

Reply:
Good catch. Looking into it.